src/NodeParser/MappedTypeNodeParser.ts

```diff
@@ -103,7 +103,7 @@ export class MappedTypeNodeParser implements SubNodeParser {
 
     protected getProperties(node: ts.MappedTypeNode, keyListType: UnionType, context: Context): ObjectProperty[] {
         return keyListType
-            .getTypes()
+            .getFlattenedTypes(derefType)
             .filter((type): type is LiteralType => type instanceof LiteralType)
             .map((type) => [type, this.mapKey(node, type, context)])
```

Comment:
```suggestion
            .reduce((acc: LiteralType[], curr: LiteralType) => {
                if (
                    acc.findIndex((val: LiteralType) => {
                        return val.getId() === curr.getId();
                    }) < 0
                ) {
                    acc.push(curr);
                }
                return acc;
            }, [])
            .map((type) => [type, this.mapKey(node, type, context)])
```

Reply:
We can use `uniqueTypeArray` here instead I think.